lib/active_merchant/billing/gateways/pay_trace.rb

```diff
@@ -52,7 +59,15 @@ def purchase(money, payment_or_customer_id, options = {})
           end
         else
           post = build_purchase_request(money, payment_or_customer_id, options)
-          post[:customer_id] ? endpoint = ENDPOINTS[:customer_id_sale] : endpoint = ENDPOINTS[:keyed_sale]
+          endpoint = if payment_or_customer_id.kind_of?(Check)
+                       ENDPOINTS[:ach_sale]
+                     elsif options[:check_transaction]
+                       ENDPOINTS[:ach_customer_sale]
+                     elsif post[:customer_id]
```

Comment:
`check_transaction` is a kind of flag that we will be using in order to identify ACH transactions as there is no way to identify ACH transactions in the case of capture, refund, or void because they get string of authorization in both cases.

`check_transaction_id ` is a field returned in the gateway response that is used as an authorization to the subsequent transactions. 

I don't think we can combine them because they both have different purpose & types. Can you please elaborate if there is any better way you are thinking for?

Reply:
Okay, that makes sense. Thank you for the extra context. It sounds like we need both fields then 👍